pkg/util/provider/metrics/metrics.go

```diff
@@ -80,6 +80,20 @@ var (
 		Help:      "Number of Failed Cloud Service API requests, partitioned by provider, and service.",
 	}, []string{"provider", "service"},
 	)
+
+	APIRequestDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Subsystem: cloudAPISubsystem,
+		Name:      "api_request_duration",
+		Help:      "Time it takes for a provider API request to complete",
+	}, []string{"provider", "service"})
```

Comment:
`virtual_machine_get`, `virtual_machine_update`, `virtual_machine_delete`, `virtual_machine_create`, `subnet_get`, `nic_get`, `nic_delete`, `nic_create` and similar for all other resource APIs. 
Using `service` label is not accurate but it was already used for other metrics like `APIRequestCount` and `APIFailedRequestCount`. 

I would rather want the following labels:
- provider
- resource_api
- status_code // Right now we only capture duration metrics for successful calls. If we wish to capture all calls then its essential to capture the returned status code. If only successful calls needs to be metered then this can be ommitted.

Reply:
since we already send the status code for the driver calls , lets introduce `status_code` for them? would help us track the number of errors due to `ResourceExhausted` also

Reply:
I think it would require two additional lables:
- `success` - to tell if the call was successful
- `status_code` - for the returned status code on an error